source/set-required.d.ts

```diff
@@ -27,8 +27,14 @@ type SomeRequired = SetRequired<Foo, 'b' | 'c'>;
 @category Object
 */
 export type SetRequired<BaseType, Keys extends keyof BaseType> =
-	Simplify<
-	BaseType &
-	// Pick the keys that should be required from the base type and make them required.
-	Required<Pick<BaseType, Keys>>
-	>;
+	// `extends unknown` is always going to be the case and is used to convert any
+	// union into a [distributive conditional
+	// type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#distributive-conditional-types).
+	BaseType extends unknown
+		? Simplify<
+		// Pick just the keys that are optional from the base type.
+		Except<BaseType, Keys> &
```

Comment:
This part reinstates the original code prior to https://github.com/sindresorhus/type-fest/pull/611